pkg/controller/cluster/controller.go

```diff
@@ -23,33 +23,90 @@ import (
 
 	"github.com/kubernetes-incubator/apiserver-builder/pkg/builders"
 
-	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
 	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
 	listers "sigs.k8s.io/cluster-api/pkg/client/listers_generated/cluster/v1alpha1"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"github.com/golang/glog"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/cluster-api/util"
 )
 
 // +controller:group=cluster,version=v1alpha1,kind=Cluster,resource=clusters
 type ClusterControllerImpl struct {
 	builders.DefaultControllerFns
 
-	// lister indexes properties about Cluster
-	lister listers.ClusterLister
+	actuator Actuator
+
+	// clusterLister holds a lister that knows how to list Cluster from a cache
+	clusterLister listers.ClusterLister
+
+	kubernetesClientSet *kubernetes.Clientset
+	clientSet           clientset.Interface
+	clusterClient       v1alpha1.ClusterInterface
 }
 
 // Init initializes the controller and is called by the generated code
 // Register watches for additional resource types here.
-func (c *ClusterControllerImpl) Init(arguments sharedinformers.ControllerInitArguments) {
-	// Use the lister for indexing clusters labels
-	c.lister = arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Clusters().Lister()
+func (c *ClusterControllerImpl) Init(arguments sharedinformers.ControllerInitArguments, actuator Actuator) {
+	cInformer := arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Clusters()
+
+	c.clusterLister = cInformer.Lister()
+
+
+	c.actuator = actuator
+
+	cs, err := clientset.NewForConfig(arguments.GetRestConfig())
+	if err != nil {
+		glog.Fatalf("error creating cluster client: %v", err)
```

Comment:
I realize we were using glog.Fatalf to control control flow in other places but I question expanding the use of it. For example, is the lifecycle of this class such that if it cannot init the program should exit? I would argue that we should be reducing the usage and relying errors bubbling back up the stack to the main.go of the program is the program should exit.

However, if this method is being called by generated / framework code and therefore we can't force the process to exit then I can see why this approach is being taken.